### DISCUSSION TITLE
Org wide path searches (with regex) do not return a list of all of the org's results

### DISCUSSION THREAD

**Discussion by n-hebert** [community/community#10039] [Other Feature Feedback, Questions, & Ideas]
When I do a search like `org:my-org path:my-path.txt /regex.* [0-9]/`, I expect to see a list of all of the files in the org that fit the path spec and the regex.
Instead, it seems to filter down to the first repository (alphabetically) that matches the search.

I want to see small snippets of all of the hits across the org, not just immediately be dumped into one file.
Is this expected behaviour?

**Comment by gorzell** [comment]
Generally, if you end up directly in a file it means that is the only result that was found.  If you know there is more than one result, then we would probably need a more concrete example so that we could reproduce it.

**Reply by n-hebert** [reply]
I'm seeing `Results are not exhaustive because query was too expensive to satisfy, consider refining your query!` when I run just the regex. It is *not* the only result, but it seems the engine can't handle much more than a few hits?

**Reply by gorzell** [reply]
That's not surprising, if you remove the `org` and `path` constraints it is going to be a very broad query.

The more important point is that if you get exactly one match (i.e. are jumped directly to a file) that means that in the content we have indexed there is only one match.  If there are other files in the org/path you have specified that you would expect to have match the query, then it would be help to have a bit more information about what those files look like, i.e. path/filename, language, etc.  As mentioned in the [FAQ](https://cs.github.com/about/faq#indexed-content) we do not index all content in a repo, so it could be that the matches you may expect to find are actually not in the index for some reason.

**Reply by n-hebert** [reply]
I'm searching for a version regex within a `versions.tf`... I don't see any reason why (if one is in the index) the others are not.

Here's the search query:
https://cs.github.com/?scopeName=All+repos&scope=&q=org%3Arewindio+path%3Aversions.tf+%2Fversion.*%3D+%22%3F3.[0-9]*\.%2F

**Reply by n-hebert** [reply]
I've been squashing things matching this pattern, so there should be more than one hit, but there may not be. A few days ago there most certainly was, as Sourcegraph was showing over a dozen.

**Reply by gorzell** [reply]
Thanks for the details, I'll have a look, although at the point at which you have finished it may be hard to figure out what was going on.

**Reply by n-hebert** [reply]
Thanks for taking a look. I'll confirm that (now that things are whittled down), there should still be 5 search results in various `versions.tf` after I finish up today(ish)

**Reply by gorzell** [reply]
We were able to figure out the root cause here, and will work on a fix.  It's not super trivial, so I don't have a timeline for when it will be ready.  Thanks again for reporting it and providing all the extra information.

**Reply by n-hebert** [reply]
Excellent! Glad to hear the bug was found and thanks for looking into it and touching back.

Happy coding

**Comment by hissain** [comment]
```
import requests

GITHUB_API_URL = "https://api.github.com/search/code"

def search_github(query, user=None, repo=None, language=None, path=None, ext=None, max_results=None, token=None):
    headers = {"Authorization": f"token {token}"}
    params = {"q": query}
    
    if user:
        params["q"] += f" user:{user}"
    if repo:
        params["q"] += f" repo:{repo}"
    if language:
        params["q"] += f" language:{language}"
    if path:
        params["q"] += f" path:{path}"
    if max_results:
        params["per_page"] = max_results

    print(f"Sending request to GitHub API with params: {params}")
    response = requests.get(GITHUB_API_URL, headers=headers, params=params)
    response.raise_for_status()
    results = response.json().get("items", [])
    print(f"GitHub API returned {len(results)} items.")
    return results
```

When I call with regular expression in path argument, the result is always returning empty. I assume the search code API supports regex in path as I found similar search in Github web interface returns some results. Is there any incorrect implementation or concept here?

**Reply by look** [reply]
@hissain sorry for the inconvenience, but the REST API does not yet support regular expressions like the web interface. For now, you need to use the [old search syntax](https://docs.github.com/en/search-github/searching-on-github/searching-code).

**Reply by hissain** [reply]
@look not even glob operators?

**Reply by look** [reply]
Correct. If it wasn't supported by the old search syntax, the API does not support it.
